pkg/client/group_mgmt_client.go

```diff
@@ -384,3 +405,62 @@ func unwrapError(body []byte) (string, error) {
 	}
 	return errResp, nil
 }
+
+// processAsyncResponse: process http code 202 response
+func processAsyncResponse(client *GroupMgmtClient, body []byte) (interface{}, error) {
+	errResp, _ := unwrapError(body)
+	if client.WaitOnJob { // check sync flag
+		unwrapMessage := &ErrorResponse{}
+		err := json.Unmarshal(body, unwrapMessage)
+		if err != nil {
+			return nil, err
+		}
+		ids, err := handleAsyncJob(unwrapMessage, client)
+
+		if err != nil {
+			return nil, fmt.Errorf("http response error: status (202), messages: %v", err.Error())
+		}
+		return ids, fmt.Errorf("http response error: status (202), messages: %v", errResp)
+	}
+	return nil, fmt.Errorf("http response error: status (202), messages: %v", errResp)
+}
+
+//handleAsyncJob : extract the job_id and monitor it periodically until job completion or timed out
+func handleAsyncJob(resp *ErrorResponse, client *GroupMgmtClient) (string, error) {
+	// filter list of async job id from the message
+
+	var objectId string
+	var jobId string
+
+	for _, msg := range resp.Messages {
+		if strings.Compare(msg.Code, smAsyncJobId) == 0 {
+			jobId = msg.Arguments.JobId
+		}
+	}
+
+	if len(jobId) == 0 {
+		return jobId, fmt.Errorf("handleAsyncJob: failed to get the job id")
+	}
+	// Loop over job ids periodically unitl 300 sec timeout or unitl completion of jobs.
+	intervalChan := time.Tick(retryFrequency) // control the fequency of GetObject() API call.
+	timeoutChan := time.After(jobTimeout)     // timeout setting, 300 Seconds
+	for {
+		select {
+		case <-intervalChan:
+
+			job, err := client.GetJobObjectSet().GetObject(jobId)
+			if err != nil {
+				fmt.Println("Warning : failed to %s jobId info, err : %s", jobId, err.Error())
```

Comment:
How about this?

  fmt.Println("Failed to get job info for ID %s, err : %s", jobId, err.Error())

Reply:
It seems like we are using the job ID as a verb, which makes it difficult for me to understand the message.

Reply:
We can fix this later. In next PR